device/pcie/pci_device.hpp

```diff
@@ -34,84 +34,144 @@ struct PciDeviceInfo
 };
 
 class PCIDevice {
+    const std::string device_path;  // Path to character device: /dev/tenstorrent/N
+    const int device_num;           // N in /dev/tenstorrent/N
```

Comment:
I'm not convinced device_num is a more clarifying name than device_id.
Is maybe pci_device_id better?

Reply:
A PCI device ID is a 16-bit identifier that uniquely identifies a PCI device model, it's part of the PCI spec.  For Grayskull it is `0xfaca`, Wormhole `0x401e`.  I was trying to avoid having "device ID" in the name because of how overloaded this term is in our system.

Reply:
I agree device_id is much more ambiguous than anything. 
What about pci_device_num? Adding pci_ prefix makes it clearer, at least for me.

Reply:
Renamed to `pci_device_num`